app/assets/javascripts/application.js

```diff
@@ -14,3 +14,106 @@
 //= require jquery_ujs
 
 //= require_tree .
+$(document).ready(function() {
+
+  var x = document.getElementById("demo");
+  var minUpdateUserDistance = 0.01 //km
+  var userId = $("div[style='display:none']").attr('id')
+  var firebaseRef = new Firebase("https://blinding-fire-43.firebaseio.com/");
+  var geo_options = {
+  enableHighAccuracy: true,
+  maximumAge        : 30000,
+  timeout           : 27000
+  };
+
+  var currentdate = new Date();
+
+
+
+
+  // Create a GeoFire index
+  var geoFire = new GeoFire(firebaseRef);
+
+
+  function getLocation() {
+      if (navigator.geolocation) {
+          navigator.geolocation.watchPosition(checkGeoFire, null, geo_options);
+      } else {
+          x.innerHTML = "Geolocation is not supported by this browser.";
+      }
+  }
+
+
+  function checkGeoFire(position) {
+    // firebaseRef.push(userId)
+
+    geoFire.get(userId).then(function(location) {
+      if (location === null) {
+        geoFire.set(userId, [position.coords.latitude, position.coords.longitude]);
+        firebaseRef.child(userId).update({'datetime': currentdate});
+        console.log('No key in database so we set the initial position')
```

Comment:
Not all browsers support `console.log`.  I see how its really useful to you all here.  There are couple things you can do.

1. Create a "shim." This basically means create an empty function if it doesn't exist so all your JS doesn't break. `console = console || {}; console.log = console.log || function() { };`
2. Create an OOJS logging object.  It would have 1 function `log` that calls `console.log`.  This way you *could* turn this off in production by passing the Rails.env into JS (see above).